test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -0,0 +1,223 @@
+			}
+
+			perfScheduling(test.Nodes.Num, test.InitPods.Num, test.PodsToSchedule.Num, nodeStrategy, setupPodStrategy, testPodStrategy, nodeSpec)
+		})
+	}
+}
+
+func perfScheduling(numNodes, numExistingPods, minPods int,
+	nodeStrategy testutils.PrepareNodeStrategy,
+	setupPodStrategy, testPodStrategy testutils.TestPodCreateStrategy,
+	nodeSpec *v1.Node) {
+
+	finalFunc, podInformer, clientset := mustSetupScheduler()
+	defer finalFunc()
+
+	var nodePreparer testutils.TestNodePreparer
+	if nodeSpec != nil {
+		nodePreparer = framework.NewIntegrationTestNodePreparerWithNodeSpec(
+			clientset,
+			[]testutils.CountToStrategy{{Count: numNodes, Strategy: nodeStrategy}},
+			nodeSpec,
+		)
+	} else {
+		nodePreparer = framework.NewIntegrationTestNodePreparer(
+			clientset,
+			[]testutils.CountToStrategy{{Count: numNodes, Strategy: nodeStrategy}},
+			"scheduler-perf-",
+		)
+	}
+
+	if err := nodePreparer.PrepareNodes(); err != nil {
+		klog.Fatalf("%v", err)
+	}
+	defer nodePreparer.CleanupNodes()
+
+	config := testutils.NewTestPodCreatorConfig()
+	config.AddStrategy(setupNamespace, numExistingPods, setupPodStrategy)
+	podCreator := testutils.NewTestPodCreator(clientset, config)
+	podCreator.CreatePods()
+
+	for {
+		scheduled, err := getScheduledPods(podInformer)
+		if err != nil {
+			klog.Fatalf("%v", err)
+		}
+		if len(scheduled) >= numExistingPods {
+			break
+		}
+		klog.Infof("got %d existing pods, required: %d", len(scheduled), numExistingPods)
+		time.Sleep(1 * time.Second)
+	}
+
+	scheduled := int32(0)
+	completedCh := make(chan struct{})
+	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(old, cur interface{}) {
+			curPod := cur.(*v1.Pod)
+			oldPod := old.(*v1.Pod)
+
+			if len(oldPod.Spec.NodeName) == 0 && len(curPod.Spec.NodeName) > 0 {
+				if atomic.AddInt32(&scheduled, 1) >= int32(minPods) {
+					completedCh <- struct{}{}
+				}
+			}
+		},
+	})
+
+	config = testutils.NewTestPodCreatorConfig()
+	config.AddStrategy(testNamespace, minPods, testPodStrategy)
+	podCreator = testutils.NewTestPodCreator(clientset, config)
+	podCreator.CreatePods()
+
+	<-completedCh
+}
+
+func getPodStrategy(pc podCase) testutils.TestPodCreateStrategy {
+	basePod := makeBasePod()
+	if pc.PodTemplatePath != nil {
+		basePod = &v1.Pod{}
+		getSpec(pc.PodTemplatePath, basePod)
+	}
+	if pc.ReplicationControllerName != nil {
+		return testutils.NewWithControllerCreatePodStrategy(*pc.ReplicationControllerName, basePod)
+	}
+	if pc.PersistentVolumeClaimTemplatePath == nil {
+		return testutils.NewCustomCreatePodStrategy(basePod)
+	}
+
+	pvTemplate := &v1.PersistentVolume{}
+	getSpec(pc.PersistentVolumeTemplatePath, pvTemplate)
+	pvcTemplate := &v1.PersistentVolumeClaim{}
+	getSpec(pc.PersistentVolumeClaimTemplatePath, pvcTemplate)
+	return testutils.NewCreatePodWithPersistentVolumeStrategy(pvcTemplate, getCustomVolumeFactory(pvTemplate), basePod)
+}
+
+func getSpec(path *string, spec interface{}) {
```

Comment:
nit: getSpec -> getSpecFromFile

Alternatively, it's more readable if you add individual helper functions such as  `func getTestCases(path string) []TestCase`, same for pod spec, node spec, etc.

Reply:
Help functions created.